### DISCUSSION TITLE
Start page number from the second page in creating PDF?

### DISCUSSION THREAD

**Discussion by kcho17** [jgm/pandoc#9846] [Q&A]
Is there any way to start page numbering from the second page (or beyond) when creating PDF from markdown document?

**Comment by TomBener** [comment]
You could see the answers [here](https://stackoverflow.com/questions/501378/how-can-i-start-pagenumbers-where-the-first-section-occurs-in-latex).

**Reply by kcho17** [reply]
Thanks for the answer. But I am trying to create PDF from markdown document, and I am not sure how to implement the solution you referred in that case. I tried to put 

```
---
\maketitle: 1
\tableofcontents: 2
\setcounter{page}{1}
\section{Introduction}: 1
...
```
In the YAML metadata in front of the md document but it didn't work.

**Reply by TomBener** [reply]
You should use `header-includes` to add extra LaTeX packages and commands. For example:

```yaml
---
header-includes: |
  \maketitle: 1
  \tableofcontents: 2
  \setcounter{page}{1}
  \section{Introduction}: 1
...
```

**Reply by kcho17** [reply]
I did add ``header-includes`` but it gives me the following error message: 
```
Error producing PDF.
! LaTeX Error: Missing \begin{document}.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.50 
```

**Reply by TomBener** [reply]
Can you post a minimal working example to diagnose the issue?

**Reply by kcho17** [reply]
Of course! Here is YAML metadata and the document I am working on:

```
---
title: AAA
author: John Doe
margin-left: 2.54cm
margin-right: 2.54cm
margin-top: 2.54cm
margin-bottom: 2.54cm
linestretch: 2
indent: true
documentclass: article
fontsize: 12pt
header-includes: |
  \renewcommand{\normalsize}{\fontsize{12pt}{14pt}\selectfont}
  \renewcommand{\footnotesize}{\fontsize{10pt}{12pt}\selectfont}
  \usepackage{titlesec}
  \newcommand{\sectionbreak}{\clearpage}
 ...
# Introduction
Pandoc is a Haskell library for converting from one markup format to another, and a command-line tool that uses this library.
```
This is the command I am using:
```
 pandoc example.md -s -o example.pdf --pdf-engine=xelatex -C --bibliography=library.bib --number-sections -N --toc=true
```
Thanks!

**Reply by tarleb** [reply]
It looks like the line containing the closing `...` for YAML is indented. Try to remove the leading space and add an empty line after the marker.

**Reply by kcho17** [reply]
I tried it but am still getting the same error message.

```
---
title: AAA
author: John Doe
margin-left: 2.54cm
margin-right: 2.54cm
margin-top: 2.54cm
margin-bottom: 2.54cm
linestretch: 2
indent: true
documentclass: article
fontsize: 12pt
header-includes: |
  \renewcommand{\normalsize}{\fontsize{12pt}{14pt}\selectfont}
  \renewcommand{\footnotesize}{\fontsize{10pt}{12pt}\selectfont}
  \usepackage{titlesec}
  \newcommand{\sectionbreak}{\clearpage}
  \maketitle: 1
  \tableofcontents: 2
  \setcounter{page}{1}
  \section{Introduction}: 1
...

# Introduction
Pandoc is a Haskell library for converting from one markup format to another, and a command-line tool that uses this library.

```
This is what it looks like now

**Reply by TomBener** [reply]
Try this instead:

````markdown
---
title: AAA
author: John Doe
margin-left: 2.54cm
margin-right: 2.54cm
margin-top: 2.54cm
margin-bottom: 2.54cm
linestretch: 2
indent: true
documentclass: article
fontsize: 12pt
header-includes: |
  \renewcommand{\normalsize}{\fontsize{12pt}{14pt}\selectfont}
  \renewcommand{\footnotesize}{\fontsize{10pt}{12pt}\selectfont}
  \usepackage{titlesec}
  \newcommand{\sectionbreak}{\clearpage}
...

# Introduction

Pandoc is a Haskell library for converting from one markup format to another, and a command-line tool that uses this library.

```{=latex}
\newpage
\setcounter{page}{1}
```

# Second page

This is the second page
````

**Reply by kcho17** [reply]
I am getting this error message now:

```
Error producing PDF.
! LaTeX Error: Can be used only in preamble.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.68 \usepackage
```

**Reply by TomBener** [reply]
It works on my side:

1. Copy the code block above.
2. Run the following command (macOS):

       pbpaste | pandoc -o test.pdf

And it managed to generate `test.pdf`. Can you double-check it?

**Reply by kcho17** [reply]
It works if I just copy the block and use the command. I am not sure what it doesn't work if I use the markdown document as input. Maybe something wrong with my command?
```
pandoc test.md -s -o test.pdf --pdf-engine=xelatex -C --bibliography=library.bib --number-sections --toc=true --toc-depth=2    
```

**Reply by kcho17** [reply]
Also, is there any way **not** to put the page number in the first page? Thanks very much for your help!

**Reply by TomBener** [reply]
You can change the output as `latex` instead of `pdf` to diagnose the issue. Also make sure to check your saved markdown file. To remove the page number for the first page, you can add `\thispagestyle{empty}`:

````md
---
title: AAA
author: John Doe
margin: 2.54cm
linestretch: 2
indent: true
documentclass: article
fontsize: 12pt
header-includes: |
  \renewcommand{\normalsize}{\fontsize{12pt}{14pt}\selectfont}
  \renewcommand{\footnotesize}{\fontsize{10pt}{12pt}\selectfont}
  \usepackage{titlesec}
  \newcommand{\sectionbreak}{\clearpage}
...

```{=latex}
\thispagestyle{empty}
```

```{=latex}
\newpage
\pagestyle{plain}
\setcounter{page}{1}
```

# Introduction

Pandoc is a Haskell library for converting from one markup format to another, and a command-line tool that uses this library.

# Second page

This is the second page
````

**Reply by kcho17** [reply]
Thanks! I checked markdown file once again now it works. But it still produce the title page with a page number (as 1); it just delete a page number for the table of contents (which I generated with the above command). Can I just delete the page number for the title page (which is now separated), not the page number for the table of contents?

**Reply by TomBener** [reply]
I don't really understand what you want to change. Do you want to change the PDF page label?

**Reply by kcho17** [reply]
I made some screenshots. Hope that it makes sense. Thanks so much for your help!
<img width="586" alt="Screenshot 2024-06-06 at 1 41 14 PM" src="https://github.com/jgm/pandoc/assets/170846864/90d18008-5417-4887-99ef-0bd46e91bcc3">
<img width="597" alt="Screenshot 2024-06-06 at 1 42 46 PM" src="https://github.com/jgm/pandoc/assets/170846864/21d68d67-0dbc-407b-b37f-397933c6300b">

**Reply by kcho17** [reply]
<img width="716" alt="Screenshot 2024-06-06 at 1 44 07 PM" src="https://github.com/jgm/pandoc/assets/170846864/3037078f-613c-4743-a3b8-d6f4b0a6a4df">

**Reply by TomBener** [reply]
The code below should meet your needs:

````yaml
---
title: AAA
author: John Doe
margin: 2.54cm
linestretch: 2
indent: true
documentclass: article
fontsize: 12pt
header-includes: |
  \renewcommand{\normalsize}{\fontsize{12pt}{14pt}\selectfont}
  \renewcommand{\footnotesize}{\fontsize{10pt}{12pt}\selectfont}
  \usepackage{titlesec}
  \newcommand{\sectionbreak}{\clearpage}
...

```{=latex}
\thispagestyle{empty}

\newpage
\pagestyle{plain}
\tableofcontents
\pagenumbering{roman}

\newpage
\pagenumbering{arabic}
```

# Introduction

Pandoc is a Haskell library for converting from one markup format to another, and a command-line tool that uses this library.

# Second page

This is the second page
````

Note the code disables `--toc` to add TOC by Pandoc, instead the TOC is added manually in the raw LaTeX code for customizing the page style.

**Reply by bpj** [reply]
You need to have a raw latex block around the LaTeX code in `header-includes` because the metadata is parsed as Markdown


``````yaml
---
header-includes: |
  ````{=latex}
  % LaTeX code goes here
  ````
---
``````

**Reply by jottr** [reply]
Four backticks? Are you sure?

**Reply by bpj** [reply]
Three bqckticks is the *minimum*. You can have more if you like, and if there are sequences of backticks inside the block you have to have a sequence with at least one more in the outer fence.

**Comment by kcho17** [comment]
I am sorry to report, but this still does't work... Maybe the title page is processed before the latex commands kicks in? 
[test.pdf](https://github.com/user-attachments/files/15612406/test.pdf)

**Reply by TomBener** [reply]
> Note the code disables `--toc` to add TOC by Pandoc, instead the TOC is added manually in the raw LaTeX code for customizing the page style.

As I have commented above, please check if you have removed `--toc` on the command line or `toc: true` in the yaml block.

**Reply by kcho17** [reply]
It now works beautifully! Thanks very much for your patience!!

**Reply by kcho17** [reply]
Sorry one more question... How can I change the linespacing for table of contents then? It looks like now everything including tables of contents is double-spaced...

**Reply by TomBener** [reply]
This [link](https://tex.stackexchange.com/questions/56546/how-to-change-spaces-between-items-in-table-of-contents) is probably helpful for you.

**Comment by ngocptblaplafla** [comment]
I wrote a filter for custom page style easier with div
[page-settings.lua](https://github.com/ngocptblaplafla/pandoc-texlive-full/blob/main/pandoc/filters/page-settings.lua)
i haven't write manual yet but code is clean and easy to understand

**Reply by jottr** [reply]
link is 404

**Reply by ngocptblaplafla** [reply]
sorry, here is new link [page-settings.lua](https://github.com/tritiumd/pandoc-filters/blob/main/page-settings.lua)
